test/Network/MQTT/EncoderSpec.hs

```diff
@@ -0,0 +1,58 @@
+{-# LANGUAGE OverloadedStrings, RecordWildCards #-}
+{-# LANGUAGE FlexibleInstances, MultiParamTypeClasses #-}
+
+module Network.MQTT.EncoderSpec (spec) where
+
+import           Network.MQTT.Packet
+import           Network.MQTT.Encoder             (encodePacket)
+import           Network.MQTT.Parser              (parsePacket)
+
+import           Test.SmallCheck.Series
+import           Test.SmallCheck
+import           Test.Hspec
+import           Test.Hspec.SmallCheck            (property)
+import           Test.SmallCheck.Series.Instances ()
+import           Data.Attoparsec.ByteString       (parseOnly)
+import           Data.ByteString.Builder          (toLazyByteString)
+import           Data.ByteString.Lazy             (toStrict)
+import           Data.Either                      (isRight)
+import           Data.Maybe                       (isNothing)
+
+spec :: Spec
+spec = do
+  describe "Parser-Encoder" $ do
+    it "parsePacket . encodePacket = id" $ property $ \p ->
+      let bytes = toStrict . toLazyByteString $ encodePacket p
+          parsedPacket = parseOnly parsePacket bytes
+      in valid parsedPacket ==> parsedPacket == Right p
```

Comment:
We should check input packet is valid. Otherwise, if `parseOnly parsePacket bytes` always fails, it will pass the check.